flask/wrappers.py

```diff
@@ -175,7 +174,10 @@ def on_json_loading_failed(self, e):
 
         .. versionadded:: 0.8
         """
-        raise BadRequest()
+        if current_app.config['DEBUG']:
```

Comment:
DEBUG is technically around all the time but I feel that we shouldn't fail if the user has somehow managed to corrupt their config => use `config.get('DEBUG', False)`.